src/process.cr

```diff
@@ -15,15 +15,17 @@ class Process
     pid_system
   end
 
-  # Returns the process group identifier of the current process.
-  def self.pgid : Int64
-    pgid(0)
-  end
+  {% if flag?(:unix) || flag?(:docs) %}
```

Comment:
These methods must exist on windows, and raise at runtime. i.e. delete this entire commit and raise inside `pgid_system` in `src/crystal/system/win32/process.cr`.

Reply:
But if one use this methods on Windows, this error will be raised only at runtime, despite being catchable a compile-time.

At least, maybe having a warning on this ones for Windows.

Reply:
@j8r This PR is mostly an internal refactor to extract the POSIX implementation. Not a place to discuss API changes.
I think we agreed to eventually remove fork and chroot, so they don't need to exist for win32. I'm not sure about the current discussion on pgid. @ysbaddaden and @jan-zajic voted to treat it like fork and chroot because it's a very POSIX-specific thing.

Reply:
`fork` is going, and will not be in 1.0 in any form. That's why it's special.

Every other API has to be identical at compile time on every platform. This was agreed in an issue, and is the status quo across the whole rest of the stdlib: see `NotImplementedError`.

Reply:
Yes, the question is: will process group id be in 1.0 or not? If I understood correctly, this is supposed to go as well.

Reply:
@straight-shoota where's the discussion on that?

Reply:
To my knowledge there has not been a dedicated discussion on that specific feature. But it has been mentioned here and in #8515. There were no objections voiced, so it seems as if everyone is on board with this.